src/builder.rs

```diff
@@ -0,0 +1,341 @@
+    pub fn connect_timeout(mut self, duration: Option<Duration>) -> Self {
+        self.tcp_config_or_default().connect_timeout = duration;
+        self
+    }
+
+    /// Set the connection write timeout duration. If `None` is specified, the
+    /// socket write calls can block indefinitely.
+    ///
+    /// If the target is currently not TCP, it will first set it.
+    pub fn write_timeout(mut self, duration: Option<Duration>) -> Self {
+        self.tcp_config_or_default().write_timeout = duration;
+        self
+    }
+
+    /// Set the number of messages that can be queued between the caller and
+    /// background threads. If too many log calls are made and the background is
+    /// too slow, this buffer will fill up. When full, calls on the current
+    /// thread will start to block.
+    ///
+    /// If the target is currently not TCP, it will first set it.
+    pub fn buffer_size(mut self, n: usize) -> Self {
+        self.tcp_config_or_default().buffer_size = n;
+        self
+    }
+
+    /// Register a static function that will be called when errors occur in the
+    /// background thread.
+    ///
+    /// If the target is currently not TCP, it will first set it.
+    pub fn background_error_handler(mut self, f: Option<fn(Error)>) -> Self {
+        self.tcp_config_or_default().background_error_handler = f;
+        self
+    }
+
+    fn tcp_config_or_default(&mut self) -> &mut TcpTarget {
+        match &mut self.target {
+            Target::Tcp(target) => target,
+            target => {
+                *target = Target::Tcp(TcpTarget::default());
+                match target {
+                    Target::Tcp(target) => target,
+                    _ => unreachable!(),
+                }
+            }
+        }
+    }
+
+    /// Set up the builder to be used with OVH's LDP service over TLS.
+    ///
+    /// This is equivalent to the following configuration:
+    /// ```rust,ignore
+    /// builder.hostname(hostname)
+    ///     .port(12202)
+    ///     .tls(true)
+    ///     .ovh_token(token)
+    ///     .null_character(true)
+    ///     .type_suffix(true)
+    /// ```
+    #[cfg(feature = "ovh-ldp")]
+    pub fn ovh_ldp(self, hostname: String, token: String) -> Self {
+        self.hostname(hostname)
+            .port(12202)
+            .tls(true)
+            .ovh_token(token)
+            .null_character(true)
+            .type_suffix(true)
+    }
+
+    /// Enable or disable automatic null character (`\0`) appending at the end
+    /// of every record. This may be required by some backends.
+    pub fn null_character(mut self, enabled: bool) -> Self {
+        self.null_character = enabled;
+        self
+    }
+
+    /// Enable or disable automatic appending type suffix to additional fields according to this [documentation](https://help.ovhcloud.com/csm/en-logs-data-platform-field-naming-conventions?id=kb_article_view&sysparm_article=KB0055662).
```

Comment:
Should this comment wrap ?

Reply:
I will do it manually, cargo fmt doesn't do it by itself.